pkcs11/src/new/functions/object_management.rs

```diff
@@ -2,25 +2,20 @@ use crate::get_pkcs11;
 use crate::new::types::function::{Rv, RvError};
 use crate::new::types::object::{Attribute, AttributeInfo, AttributeType, ObjectHandle};
 use crate::new::types::session::Session;
-use crate::new::Pkcs11;
 use crate::new::Result;
 use pkcs11_sys::*;
 use std::convert::TryInto;
 
 // Search 10 elements at a time
 const MAX_OBJECT_COUNT: usize = 10;
 
-impl Pkcs11 {
-    pub fn find_objects(
-        &self,
-        session: &Session,
-        template: &[Attribute],
-    ) -> Result<Vec<ObjectHandle>> {
+impl<'a> Session<'a> {
+    pub fn find_objects(&self, template: &[Attribute]) -> Result<Vec<ObjectHandle>> {
```

Comment:
There should be some "global" mutex to protect against this being called concurrently from different sessions. Would've said that makes it better suited for implementation on `Pkcs11` with a `Mutex` stored on it😛